src/app/Topology/Shared/utils.tsx

```diff
@@ -241,33 +243,33 @@ export class SearchExprService {
 
 export const SearchExprServiceContext = React.createContext(new SearchExprService());
 
-export const useSearchExpression = (debounceMs = 0): [string, (expr: string) => void] => {
-  const [expr, setExpr] = React.useState('');
-  const exprSvc = React.useContext(SearchExprServiceContext);
-  const _subRef = React.useRef<Subscription>();
+export const useExprSvc = (): SearchExprService => React.useContext(SearchExprServiceContext);
+
+export const DEFAULT_MATCH_EXPR_DEBOUNCE_TIME = 250; // ms
```

Comment:
Maybe I just type a little too slowly but in manually testing this with my browser's devtools open, I find that quite often I haven't actually _stopped_ typing but am just pausing or quickly proofreading what I've written, and the API request fires right as I am about to resume writing. There's a balancing act here between input latency and number of requests for sure, but I think this default could be longer. I don't think it's really worth making this into a configurable setting though.

Reply:
sure! I am actually reading and picking idea from here: https://stackoverflow.com/questions/42361485/how-long-should-you-debounce-text-input.

Reply:
Any value you have in mind? I am thinking 275-300ms?

Reply:
Sure, 300ms sounds fine. I was thinking maybe 350ms before reading through that. The current test cases we're looking at are going to have extremely low network latency compared to realistic live deployments so it's going to feel a little slower in the real world anyway. I just am also mindful that real deployments may possibly have many more targets to test against than our test scenarios, so that's where the balancing act comes in.